### PR TITLE
Switching telemetry to opt-out and disabling telemetry in CI jobs

### PR DESCRIPTION
**Pull request by popovaan**
Description: 
Added disabling telemetry scripts to CI workflows.
Switched telemetry to opt-out in MO and OVC.

Ticket: 110429

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

File: .github/workflows/mac.yml [hunk 1]
```diff
@@ -485,7 +485,17 @@ jobs:
...
+      # Disable Telemetry
+      #
+
+      - name: Disable Telemetry
+        run: |
+          mkdir -p $HOME/intel/
+          echo 0 > $HOME/intel/openvino_telemetry
+
       - name: nGraph and IE Python Bindings Tests
+
```
Comment (akashchi) [inline .github/workflows/mac.yml:498]:
```suggestion
```

Accidental?

**Review by akashchi — APPROVED** [review]
GHA-related changes LGTM.

File: .github/workflows/mac.yml (hunk 1 above)
Comment (popovaan) [reply .github/workflows/mac.yml:498]: Yes, thanks.

**Comment by ilya-lavrenov** [comment]
Do we need to perform some optimizations to eliminate gap of 4x in tests regression when telemetry is used?
I understand the for OpenVINO CI it's turned off, what about other CIs which we cannot control?
E.g. maybe we can rely on some env variables set by set and automatically disable telemetry in this case?

See https://github.com/orgs/community/discussions/26628

**Comment by popovaan** [comment]
> Do we need to perform some optimizations to eliminate gap of 4x in tests regression when telemetry is used? 

I checked the conversion time with telemetry and there are following results.

Telemetry adds time overhead, depending on number of unique operation types in model:
0.12-0.86 sec on Windows
0.03-0.05 sec on Linux 

In tests we use small synthetic [model](https://github.com/openvinotoolkit/openvino/blob/07831c9dce7418c7d28146cf7152bb8273dfeba1/tests/layer_tests/mo_python_api_tests/test_mo_convert_complex_params.py#L15)s which conversion time is less than telemetry time overhead. 
For example conversion time on Windows of this synthetic model:
0.05 sec – without telemetry
0.19 sec – with telemetry 

Other python API tests use similar 1-3 layers models. Looks like this is the main reason of degradation. On larger models degradation is less.
For example time of [bert model](https://tfhub.dev/google/bert_uncased_L-12_H-768_A-12/1) on Windows:
0.28 sec – without telemetry
0.47 sec – with telemetry

So looks like there is nothing we can do to eliminate gap of 4x in tests, as there are mostly synthetic models used.

> I understand the for OpenVINO CI it's turned off, what about other CIs which we cannot control? E.g. maybe we can rely on some env variables set by set and automatically disable telemetry in this case?
> 
> See https://github.com/orgs/community/discussions/26628

I informed other CI owners (NNCF, Notebooks) to add openvino_telemetry file, but there is no guarantee that someone won’t add new CI job with convert_model() and it will start send telemetry.
If we can handle it automatically it would be great. I can implement disabling of telemetry sending if some env variable is set.

**Comment by ilya-lavrenov** [comment]
> If we can handle it automatically it would be great. I can implement disabling of telemetry sending if some env variable is set.

Yes, let's try to rely on some known env vars used by CI services.

File: README.md [hunk 2]
```diff
@@ -128,6 +128,14 @@ OpenVINO™ Toolkit also contains several plugins which simplify loading models
 OpenVINO™ Toolkit is licensed under [Apache License Version 2.0](LICENSE).
 By contributing to the project, you agree to the license and copyright terms therein and release your contribution under these terms.
 
+## Telemetry
+OpenVINO™ collects software performance and usage data for the purpose of improving OpenVINO™ tools. This data is collected directly by OpenVINO™ or through the use of Google Analytics 4.
+You can opt-out at any time by running the command:
+
+opt_in_out --opt_out
```
Comment (rkazants) [inline README.md:135]: should we wrap this like Python code?

File: src/bindings/python/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 numpy>=1.16.6
 singledispatchmethod; python_version<'3.8'
-openvino-telemetry>=2023.1.0
+openvino-telemetry>=2023.2.0
```
Comment (rkazants) [inline src/bindings/python/requirements.txt:3]: will we provide that 2023.2 openvino package will be future-compatible with `openvino-telemetry`?

Comment (popovaan) [reply src/bindings/python/requirements.txt:3]: Yes, it was agreed to make backward-compatible releases of telemetry. Currently there are no plans of significant interface changes except from adding new parameters to constructor. All old parameters are kept.

File: README.md (hunk 2 above)
Comment (popovaan) [reply README.md:136]: Done.

File: src/bindings/python/setup.cfg [hunk 4]
```diff
@@ -13,6 +13,7 @@ setenv =
   OV_BACKEND = {env:OV_BACKEND:"CPU"}
   PYTHONPATH = {env:PYTHONPATH}
   OpenVINO_DIR = {env:OpenVINO_DIR}
+  CI = True
```
Comment (ilya-lavrenov) [inline src/bindings/python/setup.cfg:16]: do we need it?

Comment (popovaan) [reply src/bindings/python/setup.cfg:16]: Yes, openvino-ngraph-onnx-lin jobs run Python tests using tox https://github.com/openvinotoolkit/openvino/blob/137c8e896f1ab36c88378242c022e911890c6c31/.ci/openvino-onnx/Dockerfile#L76

But tox does not pass environmental variables, they need to be specified in config: https://github.com/tox-dev/tox/issues/2659. This was the reason why these jobs didn't find CI or TF_BUILD variable in environment.

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/python/setup.cfg (hunk 4 above)
Comment (ilya-lavrenov) [reply src/bindings/python/setup.cfg:16]: ok, alternatively we could add:
```
CI = {env:CI}
```
just to bypass value from env. But maybe your choice is even better to completely ignore telemetry from tests.

File: src/bindings/python/requirements.txt (hunk 3 above)
Comment (rkazants) [reply src/bindings/python/requirements.txt]: I think it makes sense to add this note and warning that we should not remove current parameters from constructor and add only optional parameters to the constructor or add new constructor interface.
This relates to the whole external API of telemetry.

**Review by rkazants — APPROVED** [review]

**Pull request merged**